### PR TITLE
LPS-105705 | Create collapsible utility | v2

### PR DESCRIPTION
**Pull request by markocikos**
Hey @jbalsas ,

This is a followup on https://github.com/wincent/liferay-portal/pull/112

Changes:
- Fixed typos.
- Replaced `<script>` with `<aui:script>`

Please review the code.

Thank you!

/cc @wincent

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js [hunk 1]
```diff
@@ -62,6 +63,7 @@ Liferay.Address = {
 	getRegions
 };
 
+Liferay.Collapse = Collapse;
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: As I mentioned, I think I'd rather not make this a global for now and require it with `aui:script require`.

Rationale is that it's easier to make it global later than to remove it.

**Comment by markocikos** [comment]
@jbalsas I added commit that adds `aui:script require`. I also capitalized collapse.es.js file for consistency, as it contains a class.

File: modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp [hunk 2]
```diff
@@ -67,4 +67,10 @@
 		%>
 
 	</div>
+
+	<aui:script require="frontend-js-web/liferay/Collapse.es as Collapse">
```
Comment (jbalsas) [inline modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp]: Does this actually work? I wouldn't bet on this dependency being properly resolved.

Comment (jbalsas) [reply modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp]: If it does, it should be by coincidence and through the bridge-generator which we should not rely onto...

Best approach is normally to create a local module in the bundle and from there `import {Collapse} from 'frontend-js-web';` after properly exposing Collapse in the `frontend-js-web` API and properly defining it as a dependency in this bundle

Comment (markocikos) [reply modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp]: I just did an `ant all` on branch to triple-check it, it works. We are using this same import method in 11 places, [this](https://github.com/liferay/liferay-portal/blob/a2813d27c5ca56c8683af6dfbaac076fb2f4fc5e/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_shortcut.jsp#L68) for example.

Comment (jbalsas) [reply modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp]: Please, see my above reference to the `bridge-generator`

Comment (markocikos) [reply modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/product_menu.jsp]: I'll rework it to the proper solution.

**Pull request closed without merging**